satpy/etc/readers/datatailor_nc.yaml

```diff
@@ -0,0 +1,21 @@
+reader:
+  name: datatailor_nc
+  short_name: EUMETSAT Data Tailor NetCDF
+  long_name:  Generic reader for simplified NetCDF files coming from the EUMETSAT Data Tailor
+  description: >
+    The EUMETSAT Data Tailor is a tool published by EUMETSAT that allows tailoring
+    satellite data to and from various file formats, among other things.
+    This is a generic reader for the simplified NetCDF output format.
+    For more information on the data tailor, see https://www.eumetsat.int/data-tailor
+  status: Alpha
+  supports_fsspec: True
+  reader: !!python/name:satpy.readers.yaml_reader.FileYAMLReader
+  sensors: [many]
+  default_datasets:
+
+file_types:
+  datatailor_simple_nc:
+    file_reader: !!python/name:satpy.readers.datatailor_nc.DataTailorNC
+    file_patterns:
+      - '{product_key}{start_sensing_time}{stop_sensing_time}epct{process_uuid}_{steps_acronym}.nc'
+      - "{stem}.nc"
```

Comment:
This could conflict with the satpy nc reader which also accepts everything...

Reply:
It does. It means that if a user has a file `foo.nc`, that Satpy cannot automatically determine the reader. Maybe such a limitation is acceptable — that if the filename is not precise enough, that the user must provide the reader?

So far Satpy takes whatever is the first possible reader that fits. That's understandable from an optimisation point of view, but it's not very stable and violates the principle "In the face of ambiguity, refuse the temptation to guess" (Zen of Python).

Reply:
I the filename output by the data tailor really free or random?

Reply:
It depends. When using the Python API like I do, it outputs a stream. There is a server-side filename in the metadata, but of course users can name the file what they want client-side. The command-line client does create a file with the same name as on the server.

The server-side filename in the metadata follows the first of the file patterns that I indicated (source: answer from EUMETSAT Secretariat Helpdesk).  Example filename:

`IASISND02_20230602060852Z_20230602075052Z_epct_f2846918_F.nc`

I'm not too keen on forcing this, in particular due to the `process_uuid`, which is different even if the same file is downloaded with the same customisation multiple times. The other problem is that this filename pattern is not publicly documented and there is no public API to obtain the filename except by generating the product. It might be just  an implementation detail and subject to change.

(It also seems the filename pattern I was given by the Help Desk does not actually match the files I have obtained so far.)

Reply:
Ok, so nothing much we can do here when using the api to force the filename. One cool solution though would be to pass the stream directly to satpy somehow (without going through a filename) but I have not thought of the implications).

Back to filenames, so what I'm thinking right now is actually to have satpy raise an error when there is a filename that could fit multiple readers to force the user into specifying the reader to use. Would that be possible?